controllers/virtualmachine/virtualmachine/virtualmachine_controller.go

```diff
@@ -288,8 +289,10 @@ func (r *Reconciler) Reconcile(ctx context.Context, req ctrl.Request) (_ ctrl.Re
 	ctx = pkgcfg.JoinContext(ctx, r.Context)
 	ctx = cource.JoinContext(ctx, r.Context)
 
-	if pkgcfg.FromContext(ctx).Features.BringYourOwnEncryptionKey {
+	if pkgcfg.FromContext(ctx).Features.BringYourOwnEncryptionKey ||
+		pkgcfg.FromContext(ctx).Features.FastDeploy {
 		ctx = vmconfig.WithContext(ctx)
+		ctx = vmconfig.Register(ctx, diskpromo.New())
 		ctx = vmconfig.Register(ctx, crypto.New())
```

Comment:
Do  we want to guard the Register() calls based on their corresponding feature flag?

Reply:
Yes, done